### ISSUE TITLE
Proxy client connections hanging

### ISSUE THREAD

**Issue by domsj** [openvstorage/alba#461]
This happened on @dejonghb's env 

Setup: accelerated backend

Around 10:04 the node hosting (half of the) asds of the fragment cache backend went down.

In the log of the proxy of the main backend we then see this:

```
Nov 25 10:05:41 ftcmp02 alba[22625]: 2016-11-25 10:05:41 368610 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442472 - info - server: new client connection from (172.22.186.32,35092)
Nov 25 10:05:42 ftcmp02 alba[22625]: 2016-11-25 10:05:42 193100 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442473 - info - server: new client connection from (172.22.186.32,35136)
Nov 25 10:05:45 ftcmp02 alba[22625]: 2016-11-25 10:05:45 72825 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442474 - info - server: new client connection from (172.22.186.32,35590)
Nov 25 10:05:45 ftcmp02 alba[22625]: 2016-11-25 10:05:45 73584 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442475 - info - server: new client connection from (172.22.186.32,35592)
Nov 25 10:05:57 ftcmp02 alba[22625]: 2016-11-25 10:05:57 369509 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442476 - info - server: new client connection from (172.22.186.32,37264)
Nov 25 10:05:58 ftcmp02 alba[22625]: 2016-11-25 10:05:58 193949 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442477 - info - server: new client connection from (172.22.186.32,37434)
Nov 25 10:06:01 ftcmp02 alba[22625]: 2016-11-25 10:06:01 73753 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442478 - info - server: new client connection from (172.22.186.32,37904)
```

and a bit later it turns in to

```
Nov 25 10:15:50 ftcmp02 alba[22625]: 2016-11-25 10:15:50 104537 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442600 - info - server: new client connection from (172.22.186.32,40850)
Nov 25 10:15:50 ftcmp02 alba[22625]: 2016-11-25 10:15:50 105027 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442601 - info - server: new client connection from (172.22.186.32,40852)
Nov 25 10:16:02 ftcmp02 alba[22625]: 2016-11-25 10:16:02 400797 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442602 - warning - Denying connection from (172.22.186.32,43148), too many client connections 128
Nov 25 10:16:03 ftcmp02 alba[22625]: 2016-11-25 10:16:03 223398 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442603 - warning - Denying connection from (172.22.186.32,43448), too many client connections 128
Nov 25 10:16:04 ftcmp02 alba[22625]: 2016-11-25 10:16:04 401951 +0100 - ftcmp02 - 22625/0 - alba/proxy - 442604 - warning - Denying connection from (172.22.186.32,43556), too many client connections 128
```

There are a lot of open fds for deleted files:

```
root@ftcmp02:~# lsof -p 22625 | grep deleted | wc
    127    1270   19050
```

this indicates hanging client threads.
Probably caused by those asds becoming unavailable?

**Comment by wimpers** [comment]
@redlicha could you see what can be done on the voldrv/edge level to see if the voldrv/edge can try other options (like getting IO from the HDD backend.

**Comment by wimpers** [comment]
@jeroenmaelbrancke can we a check for  "too many client connections " on the monitoring side. Please create a ticket please on the correct repo

**Comment by domsj** [comment]
Seen again on the env of @dejonghb 

In the lsof output we saw that all the fds were followed by 'w', this indicates the file was (only) opened for writing.
This means that something is hanging in the full object download path.

**Comment by toolslive** [comment]
#740

**Comment by domsj** [comment]
probably fixed by #793, create a new ticket or reopen in case this would be seen again
